dlt/common/cli/runner/inquirer.py

```diff
@@ -0,0 +1,175 @@
+    """
+
+    def __init__(self, params: RunnerParams, pipeline_members: PipelineMembers) -> None:
+        self.params = params
+        self.pipelines = pipeline_members.get("pipelines")
+        self.sources = pipeline_members.get("sources")
+
+    def maybe_ask(self) -> t.Tuple[dlt.Pipeline, t.Union[DltResource, DltSource]]:
+        """Shows prompts to select pipeline, resources and sources
+
+        Returns:
+            (DltResource, DltSource): a pair of pipeline and resources and sources
+        """
+        self.preflight_checks()
+        pipeline_name = self.get_pipeline_name()
+        pipeline = self.pipelines[pipeline_name]
+
+        source_name = self.get_source_name()
+        resource = self.sources[source_name]
+        if isinstance(resource, DltResource):
+            label = "Resource"
+        else:
+            label = "Source"
+
+        real_source_name = ""
+        if resource.name != source_name:
+            real_source_name = f" ({resource.name})"
+
+        real_pipeline_name = ""
+        if pipeline.pipeline_name != pipeline_name:
+            real_pipeline_name = f" ({pipeline.pipeline_name})"
+
+        fmt.echo("\nPipeline: " + fmt.style(pipeline_name + real_pipeline_name, fg="blue"))
+
+        fmt.echo(f"{label}: " + fmt.style(source_name + real_source_name, fg="blue"))
+        return pipeline, resource
+
+    def get_pipeline_name(self) -> str:
+        if self.params.pipeline_name:
+            return self.params.pipeline_name
+
+        if len(self.pipelines) > 1:
+            message, options, values = make_select_options("pipeline", self.pipelines)
+            choice = self.ask(message, options, default="n")
+            pipeline_name = values[choice]
+            return pipeline_name
+
+        pipeline_name, _ = next(iter(self.pipelines.items()))
+        return pipeline_name
+
+    def get_source_name(self) -> str:
+        if self.params.source_name:
+            return self.params.source_name
+
+        if len(self.sources) > 1:
+            message, options, values = make_select_options("resource or source", self.sources)
+            choice = self.ask(message, options, default="n")
+            source_name = values[choice]
+            return source_name
+
+        source_name, _ = next(iter(self.sources.items()))
+        return source_name
+
+    def ask(self, message: str, options: t.List[str], default: t.Optional[str] = None) -> int:
+        choice = fmt.prompt(message, options, default=default)
+        if choice == "n":
+            raise FriendlyExit()
+
+        return int(choice)
+
+    def preflight_checks(self) -> None:
+        if pipeline_name := self.params.pipeline_name:
+            if pipeline_name not in self.pipelines:
+                fmt.echo(
+                    fmt.error_style(
+                        f"Pipeline {pipeline_name} has not been found in pipeline script"
+                        "You can choose one of: "
```

Comment:
![image](https://github.com/dlt-hub/dlt/assets/20367975/9307fa60-581f-4396-9be3-32c50de665ad)

Reply:
I can only run pipelines that I've already run?

Reply:
can you provide here more info? e.g. "Ensure you have already run the pipeline with `dlt run <pipeline_script>.py`"
also check len(self.pipelines.keys()) > 0